pep-0654.rst

```diff
@@ -469,7 +469,13 @@ recursively, as follows:
         else:
             # exc is a leaf exception and its traceback
             # is the concatenation of the traceback
-            # segments in tbs
+            # segments in tbs.
+
+            # Note: the traceback segments are shallow-copied
+            # into tbs so they can be shared by multiple leaf
+            # exceptions. Make a deep copy if your use case
+            # requires that.
```

Comment:
Huh, this is actually ambiguous (I hadn't realized this before): the singly-linked list of Traceback objects is mutable (you could set any item's `tb_next` and mutate the chain) and in addition the `tbs` list is reused by each iteration through the generator.

I'm not particularly worried about people mutating the Traceback objects, but somehow your comment appears to refer to them. To describe the reuse of `tbs`, maybe reformulate the note like this?

> Note: the list returned (`tbs`) is reused by each iteration through the generator.
> Make a copy of your use case holds on to it beyond the current iteration.

Reply:
Ah I see, yes. I am a bit concerned also about people concatenating the tb chunks of tbs to make it look like one traceback.